From 94ec618bd1a6b07fafbbfc9bcc54e7f9360ff9a0 Mon Sep 17 00:00:00 2001
From: Florian Fainelli <florian@openwrt.org>
Date: Mon, 28 Jan 2013 20:06:23 +0100
Subject: [PATCH 05/11] MIPS: BCM63XX: introduce BCM63XX_OHCI configuration
 symbol

This configuration symbol can be used by CPUs supporting the on-chip
OHCI controller, and ensures that all relevant OHCI-related
configuration options are correctly selected. So far, OHCI support is
available for the 6328, 6348, 6358 and 6358 SoCs.

Signed-off-by: Florian Fainelli <florian@openwrt.org>
---
 arch/mips/bcm63xx/Kconfig |   15 ++++++++++-----
 1 file changed, 10 insertions(+), 5 deletions(-)

--- a/arch/mips/bcm63xx/Kconfig
+++ b/arch/mips/bcm63xx/Kconfig
@@ -7,10 +7,17 @@ config BCM63XX_CPU_3368
 	select SYS_HAS_CPU_BMIPS4350
 	select HAVE_PCI
 
+config BCM63XX_OHCI
+	bool
+	select USB_ARCH_HAS_OHCI
+	select USB_OHCI_BIG_ENDIAN_DESC if USB_OHCI_HCD
+	select USB_OHCI_BIG_ENDIAN_MMIO if USB_OHCI_HCD
+
 config BCM63XX_CPU_6328
 	bool "support 6328 CPU"
 	select SYS_HAS_CPU_BMIPS4350
 	select HAVE_PCI
+	select BCM63XX_OHCI
 
 config BCM63XX_CPU_6338
 	bool "support 6338 CPU"
@@ -25,21 +32,25 @@ config BCM63XX_CPU_6348
 	bool "support 6348 CPU"
 	select SYS_HAS_CPU_BMIPS32_3300
 	select HAVE_PCI
+	select BCM63XX_OHCI
 
 config BCM63XX_CPU_6358
 	bool "support 6358 CPU"
 	select SYS_HAS_CPU_BMIPS4350
 	select HAVE_PCI
+	select BCM63XX_OHCI
 
 config BCM63XX_CPU_6362
 	bool "support 6362 CPU"
 	select SYS_HAS_CPU_BMIPS4350
 	select HAVE_PCI
+	select BCM63XX_OHCI
 
 config BCM63XX_CPU_6368
 	bool "support 6368 CPU"
 	select SYS_HAS_CPU_BMIPS4350
 	select HAVE_PCI
+	select BCM63XX_OHCI
 endmenu
 
 source "arch/mips/bcm63xx/boards/Kconfig"
